### DISCUSSION TITLE
Filament right for creating financial reports?

### DISCUSSION THREAD

**Discussion by madsem** [filamentphp/filament#2126] [Help] [answered]
Hey there,

Filament looks great, awesome that you put this out there open source :)

I'm tasked with building a reporting UI with user roles & permissions.

The app should download reporting data from third party APIs, save the data as Eloquent model and then create various reports with filters.

Reporting tables should have rollups at the end (total numbers)

Does this sound like filament would be the right tool for this job?

**Comment by Z3d0X** [comment, marked as answer]
You will only have to work this part out: `The app should download reporting data from third party APIs`.
As for other things mentioned filament can easily handle it

**Reply by madsem** [reply]
Hey @ZedoXmv thanks for your reply, appreciate it :) 

I've read more into Filament and I can't seem to find anything about total rows in tables (roll ups).
Is that also possible?

IE: You have a table with Order Values for example, and at the end of the table for each page you have a row with total order amount etc.

Is that doable without reinventing the wheel? :)

Thank you 🙏

**Reply by Z3d0X** [reply]
I believe this is what you are referring to
![image](https://user-images.githubusercontent.com/75579178/162893908-dcca6f5c-b649-43c0-9482-d929b3726749.png)

i recommend checking out the demo
https://demo.filamentphp.com

**Reply by madsem** [reply]
I was actually referring to totals of financial values, like in the demo on the "Orders" view.

I need to be able to add a row at the bottom with total revenue, total orders etc. 

In the demo I can only see total pages:
![D5F47273-47C6-4E99-B6B8-6DE5B0883B11](https://user-images.githubusercontent.com/3166991/162894767-6a6e3037-5b91-4a3f-b42c-fe2ca9aa7020.png)

So essentially a `SUM` or `AVG` for specific columns with numerical values

**Reply by Z3d0X** [reply]
Not sure if you can do it inside the table, but you can add `footerWidget` to show the required info
https://filamentphp.com/docs/2.x/admin/pages#building-widgets

**Reply by madsem** [reply]
ohh sweet, that looks like it could work, thank you 😊
